packages/roosterjs-editor-dom/lib/utils/adaptFontColorToBackgroundColor.ts

```diff
@@ -0,0 +1,87 @@
+const enum ColorTones {
+    BRIGHT,
+    DARK,
+}
+const LiteralDarkColor = ['black', '--ms-color-black'];
+const LiteralBrightColor = ['white', '--ms-color-white'];
+const WHITE = '#ffffff';
+const BLACK = '#000000';
+
+//Using the HSL (hue, saturation and lightness) representation for RGB color values, if the value of the lightness is less than 20, the color is dark
+const DARK_COLORS_LIGHTNESS = 20;
+//If the value of the lightness is more than 80, the color is bright
+const BRIGHT_COLORS_LIGHTNESS = 80;
+
+/**
+ * @internal
+ * Change the font color to white or some other color, so the text can be visible with a darker background
+ * @param element The element that contains text.
+ * @param backgroundColor The backgroundColor of a element
+ */
+export default function adaptFontColorToBackgroundColor(
+    element: HTMLElement,
+    backgroundColor: string
+) {
+    if (element.firstElementChild?.hasAttribute('style')) {
+        return;
+    }
+    if (isADarkOrBrightColor(backgroundColor) === ColorTones.DARK) {
+        element.style.color = WHITE;
+    } else if (isADarkOrBrightColor(backgroundColor) === ColorTones.BRIGHT) {
+        element.style.color = BLACK;
+    } else {
+        element.style.color = '';
+    }
+}
+
+function isADarkOrBrightColor(color: string) {
+    let lightness = 50; // set 50, because is not bright or dark
+    if (isRGB(color) || isRGBA(color) || isHEX(color)) {
+        lightness = calculateLightness(color);
+    }
+    if (lightness < DARK_COLORS_LIGHTNESS || LiteralDarkColor.indexOf(color) > -1) {
+        return ColorTones.DARK;
+    } else if (lightness > BRIGHT_COLORS_LIGHTNESS || LiteralBrightColor.indexOf(color) > -1) {
+        return ColorTones.BRIGHT;
+    }
```

Comment:
I think a better way is to use getComputedStyle since that will always give you rgb/rgba color (at least from what I see)

Reply:
getComputedStyle will return the light gray color for when the element is selected

Reply:
Using getPropertyValue will always return RGB or RGBA